compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -31,45 +33,75 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_stmt_without_recovery(&mut self) -> PResult<'a, Option<Stmt>> {
-        maybe_whole!(self, NtStmt, |x| Some(x));
```

Comment:
`#[attr] $stmt` wasn't supported at all previously, so this PR extends the language slightly.
Although this is consistent with `#[attr] $item`, which is already accepted.

Reply:
Yeah, I did this for consistency with items. I don't think it's needed for the rest of the PR, so I could split it out and FCP it separately.

Reply:
Nah, this seems small enough to land as a bugfix.

(The expand/builtin-macros changes look independent from parsing/token-collection though and less tricky, so it would be nice to split them to a separate PR, together with all the test additions.)

Reply:
@petrochenkov: I've opened https://github.com/rust-lang/rust/pull/79326. I've left the tests in this PR since they otherwise produce no useful output (we get a parse error before the macros are invoked).